lib/python/Screens/FlashManager.py

```diff
@@ -0,0 +1,600 @@
+				self.setIndex = self["list"].getSelectedIndex()
+				self.imagesList = {}
+				self.getImagesList()
+			except (IOError, OSError) as err:
+				self.session.open(MessageBox, _("Error %d: Unable to delete downloaded image '%s'!  (%s)" % (err.errno, currentSelection, err.strerror)), MessageBox.TYPE_ERROR, timeout=3)
+
+	def selectionChanged(self):
+		currentSelection = self["list"].l.getCurrentSelection()[0]
+		self["key_blue"].setText("" if "://" in currentSelection[1] or currentSelection[1] in ["Expanded", "Loading"] else _("Delete Image"))
+		if currentSelection[1] == "Loading":
+			self["key_green"].setText("")
+		else:
+			if currentSelection[1] == "Expanded":
+				self["key_green"].setText(_("Collapse") if currentSelection[0] in self.expanded else _("Expand"))
+				self["description"].setText("")
+			else:
+				self["key_green"].setText(_("Flash Image"))
+				self["description"].setText(_("Location: %s") % currentSelection[1][:currentSelection[1].rfind(sep) + 1])
+
+	def keyTop(self):
+		self["list"].instance.moveSelection(self["list"].instance.moveTop)
+		self.selectionChanged()
+
+	def keyPageUp(self):
+		self["list"].instance.moveSelection(self["list"].instance.pageUp)
+		self.selectionChanged()
+
+	def keyUp(self):
+		self["list"].instance.moveSelection(self["list"].instance.moveUp)
+		self.selectionChanged()
+
+	def keyDown(self):
+		self["list"].instance.moveSelection(self["list"].instance.moveDown)
+		self.selectionChanged()
+
+	def keyPageDown(self):
+		self["list"].instance.moveSelection(self["list"].instance.pageDown)
+		self.selectionChanged()
+
+	def keyBottom(self):
+		self["list"].instance.moveSelection(self["list"].instance.moveEnd)
+		self.selectionChanged()
+
+
+class FlashImage(Screen):
+	skin = """
+	<screen position="center,center" size="640,150" flags="wfNoBorder" backgroundColor="#54242424">
+		<widget name="header" position="5,10" size="e-10,50" font="Regular;40" backgroundColor="#54242424"/>
+		<widget name="info" position="5,60" size="e-10,130" font="Regular;24" backgroundColor="#54242424"/>
+		<widget name="progress" position="5,e-39" size="e-10,24" backgroundColor="#54242424"/>
+	</screen>"""
+
+	def __init__(self, session, imagename, source):
+		Screen.__init__(self, session)
+		self.containerbackup = None
+		self.containerofgwrite = None
+		self.getImageList = None
+		self.downloader = None
+		self.source = source
+		self.imagename = imagename
+		self["header"] = Label(_("Backup settings"))
+		self["info"] = Label(_("Save settings and EPG data"))
+		self["summary_header"] = StaticText(self["header"].getText())
+		self["actions"] = HelpableActionMap(self, ["OkCancelActions"], {
+			"cancel": (self.abort, _("Add help text")),
+			"ok": (self.ok, _("Add help text"))
+		}, prio=-1)
+		self["progress"] = ProgressBar()
+		self["progress"].setRange((0, 100))
+		self["progress"].setValue(0)
+		self.delay = eTimer()
+		self.delay.callback.append(self.confirmation)
+		self.delay.start(0, True)
+		self.hide()
```

Comment:
Why has
	self.callLater(self.confirmation)
been removed and been replaced with functionally equivalent code?
	self.delay = eTimer()
	self.delay.callback.append(self.confirmation)
	self.delay.start(0, True)

This necessitates the inclusion of eTimer() into this code when it is not really required.

Reply:
Check line 517